### ISSUE TITLE
Modify the socket cookie on the fly

### ISSUE THREAD

**Issue by rafinskipg** [balderdashy/sails#2646]
Hi, I do need some help with this..

Sails version 0.10.5

Relates to #1026 

I've been struggling a few days trying to find in the documentation or source code how can I change the cookie on a socket connection

The feature I'm trying to implement is the "remember me" functionality, when a user "clicks" remember me, the cookie should have a max age, otherwise it will be a session cookie.

In most of the issues you refer to an old `authorization` file that no longer exists.

Also the comments in the `socket.js` configuration files wont help me so much,

I have tried to modify the cookie on the server by accesing:

req.session.cookie = 'cookie=?..',
handshake.cookie 
handshake.headers.cookie
handshake.session.cookie
req.headers.cookie
req.cookie

I don't really know how to modify it, and the documentation is not helping me with an example on how to do it. Also, how can I access the `sails.lib.hooks.session` methods for parsing/signing cookies from a sails controller??

Thank you !

**Comment by CWyrtzen** [comment]
@rafinskipg is this still causing problems for you? are you still using v0.10.5

**Comment by sailsbot** [comment]
Thanks for posting, @rafinskipg.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
